File: src/main/java/um/haberes/core/hexagonal/cursos/curso/domain/model/Curso.java
package um.haberes.core.hexagonal.cursos.curso.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import um.haberes.core.hexagonal.facultad.domain.model.Facultad;
import um.haberes.core.hexagonal.geografica.domain.model.Geografica;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Curso {

    private Long cursoId;

    @Builder.Default
    private String nombre = "";

    private Integer facultadId;

    private Integer geograficaId;

    @Builder.Default
    private Byte anual = 0;

    @Builder.Default
    private Byte semestre1 = 0;

    @Builder.Default
    private Byte semestre2 = 0;

    private Integer nivelId;

    @Builder.Default
    private Byte adicionalCargaHoraria = 0;

    private Facultad facultad;

    private Geografica geografica;
}

File: src/main/java/um/haberes/core/kotlin/model/Curso.kt
package um.haberes.core.kotlin.model

import jakarta.persistence.*
import um.haberes.core.hexagonal.facultad.infrastructure.persistence.entity.FacultadEntity
import um.haberes.core.hexagonal.geografica.infrastructure.persistence.entity.GeograficaEntity

@Entity
@Table
data class Curso(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cursoId: Long? = null,

    var nombre: String = "",
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var anual: Byte = 0,
    var semestre1: Byte = 0,
    var semestre2: Byte = 0,
    var nivelId: Int? = null,
    var adicionalCargaHoraria: Byte = 0,

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: FacultadEntity? = null,

    @OneToOne
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: GeograficaEntity? = null,

    @OneToOne
    @JoinColumn(name = "nivelId", insertable = false, updatable = false)
    var nivel: Nivel? = null

) : Auditable()
